File: Node.js/Bubble.js
function Bubble(numbers) {

    let temp = 0;

    for(let i = 0; i < numbers.length; i++){

        for(let j = 0; j < numbers.length - i; j++){

            if(numbers[j] > numbers[j + 1]){

                temp = numbers[j];

                numbers[j] = numbers[j + 1];

                numbers[j + 1] = temp;

            }

        }

    }

}

numbers = [];

// require('fs').readFileSync('/Users/henrique_buzin/WebstormProjects/Node.js/100.000.txt',).toString().split('\n').forEach(function (line) {
// require('fs').readFileSync('/Users/henrique_buzin/WebstormProjects/Node.js/10.000.txt',).toString().split('\n').forEach(function (line) {
require('fs').readFileSync('/Users/henrique_buzin/WebstormProjects/Node.js/1.000.txt',).toString().split('\n').forEach(function (line) {

    numbers.push(parseInt(line, 10));

});

let inicio = process.hrtime();
inicio = inicio[1];

let inicioRuntime = process.memoryUsage().heapUsed;

Bubble(numbers);

let fim = process.hrtime();
fim = fim[1];

let fimRuntime = process.memoryUsage().heapUsed;

for(let i = 0; i < numbers.length; i++){

    console.log(numbers[i]);

}

console.log(fim - inicio);

console.log(fimRuntime - inicioRuntime);

File: Python/src/Bubble.py
import time

import resource

def Bubble(numbers):

    for i in range(len(numbers) - 1, 0, -1):

        for j in range(i):

            if numbers[j] > numbers[j + 1]:

                temp = numbers[j]

                numbers[j] = numbers[j + 1]

                numbers[j + 1] = temp

numbers = []

runtimex = resource.getrusage(resource.RUSAGE_SELF).ru_maxrss

# with open("/Users/henrique_buzin/PycharmProjects/Python/src/1.000.txt") as file:
# with open("/Users/henrique_buzin/PycharmProjects/Python/src/10.000.txt") as file:
with open("/Users/henrique_buzin/PycharmProjects/Python/src/1.000.txt") as file:

    for i, line in enumerate(file):

        numbers.insert(i, int(line.strip("\n")))

inicio = time.time_ns()

inicioRuntime = resource.getrusage(resource.RUSAGE_SELF).ru_maxrss

Bubble(numbers)

fim = time.time_ns()

fimRuntime = resource.getrusage(resource.RUSAGE_SELF).ru_maxrss

print(numbers)

print(fim - inicio)

print(fimRuntime - inicioRuntime)
